### ISSUE TITLE
Windows support?

### ISSUE THREAD

**Issue by zkat** [mfontanini/presenterm#13]
I noticed this compiles on Windows and you even build a Windows release, but when I try to run a presentation it says `Failed to run presentation: io: Window pixel size not implemented for Windows API`. This happens in Windows Terminal, WezTerm, and VSCode console, all on win11

**Comment by mfontanini** [comment]
Looks like crossterm doesn't know how to find the terminal window size in pixels. I can patch this by avoiding getting the window size and only getting the terminal size in columns/rows easily but then you won't be able to render images as I need the size in pixels to be able to scale it correctly (and I believe the crate I use for image rendering does too). I don't use windows but if you have any pointers on how to get the window size, that'd be great.

**Comment by mfontanini** [comment]
@zkat can you pull the latest code and test it now? This will gracefully explode once you land on a slide that contains an image but in theory the rest should work. Please let me know otherwise.

We still need to figure out the screen size to claim full windows support though, so let's keep this ticket open.

**Comment by mfontanini** [comment]
Yes, can you please check with the latest source?

**Comment by zkat** [comment]
Can confirm that things work now.

**Comment by mfontanini** [comment]
I just pushed a fix that makes images work in Windows under wezterm. Other terminals may or may not work but at least I removed the hard failure this had before. I'll look into other terminals soon.

**Comment by mfontanini** [comment]
Closing this. If there's terminals where images don't work, please create a new issue.
